### ISSUE TITLE
SNOW-1462014: Ability to disable OTel span creation even if opentelemetry is installed

### ISSUE THREAD

**Issue by mikeldking** [snowflakedb/snowpark-python#1730]
## What is the current behavior?

If opentelemetry is installed, spans are automatically emitted via https://github.com/snowflakedb/snowpark-python/blob/29b3bba9a7bd13206221b6fc6ad9ed0fae853fe0/src/snowflake/snowpark/_internal/open_telemetry.py

## What is the desired behavior?

Add the ability to disable span creation even if opentelemetry is installed so as to keep the trace hierarchy simpler.

## How would this improve `snowflake-snowpark-python`?

Make it more configurable and more composable with other OTel instrumentations

## References, Other Background
OTel gives you the ability to turn off different instrumentations like so: https://github.com/snowflakedb/snowpark-python/blob/29b3bba9a7bd13206221b6fc6ad9ed0fae853fe0/src/snowflake/snowpark/_internal/open_telemetry.py

**Comment by sfc-gh-yuwang** [comment]
Hi, do you just want to be able to disable all snowpark emit spans, or you also want to have control on what function should not emit span? For example, do you want snowpark to emit span on .collect() while not emitting span on .show() ?

**Comment by mikeldking** [comment]
@sfc-gh-yuwang thanks so much for responding! For now I think it's an overall switch-off (@trevor-laviale-arize might be able to chime in here)

**Comment by trevor-laviale-arize** [comment]
Yeah I think just the ability to disable in general is good for now :)

**Comment by sfc-gh-yuwang** [comment]
Got it, will do this recently, I am thinking about having a environment variable to turn on/off snowpark opentelemetry

**Comment by trevor-laviale-arize** [comment]
Amazing, thanks so much!

**Comment by mikeldking** [comment]
Thanks @sfc-gh-yuwang - env var makes a ton of sense

**Comment by sfc-gh-yuwang** [comment]
I want to do a quick update on this issue, we are now discussing with other teams who potentially could have the issue about a uniform solution across snowflake, we should have this solved within this month

**Comment by sfc-gh-jfreeberg** [comment]
@mikeldking , @trevor-laviale-arize  -- Have you tried setting this env var? `OTEL_PYTHON_DISABLED_INSTRUMENTATIONS` ?

https://opentelemetry.io/docs/zero-code/python/configuration/#disabling-specific-instrumentations

**Comment by sfc-gh-yuwang** [comment]
I also find that opentelemetry has recently added a feature of trace config: https://github.com/open-telemetry/opentelemetry-specification/pull/3877 , can you also try it?

**Comment by mikeldking** [comment]
Sorry very late to this request. This wasn't specifically for me but something observed by one of our customers. I believe the above would work and thus I will close my request. Thanks for the help
